flist: invalid mount point while trying to deploy k8s

I got this error while I was trying to deploy k8s 
```
deploying kubernetes master on node: BDhpLyV28Y6Njy54WnB2Rkn6eHdEUbMu3GwTTs65rGAG, iprange: 172.21.10.0/24
deploying kubernetes worker on node: A7FmQZ72h7FzjkJMGXmzLDFyfyxzitDZYuernGG97nv7, iprange: 172.21.11.0/24
deploying kubernetes worker on node: 2m3nHPSAMyZFSeg5HPozic2NGBMtrXrBkhtNcVmd5Ss6, iprange: 172.21.12.0/24
Workload 481837 failed to deploy due to error could not mount k3os flist: invalid mount point: failed to check mountpoint: /var/cache/modules/flistd/mountpoint/481837-1: stat /var/cache/modules/flistd/mountpoint/481837-1: transport endpoint is not connected. For more details: https://explorer.testnet.grid.tf/api/v1/reservations/workloads/481837
```
https://explorer.testnet.grid.tf/api/v1/reservations/workloads/481837

Confirmed! the only possible way this happens is that the 0-fs process has crashed. 
I checked the logs and I got this 
```
cat 481837-1.log 
mount starts
panic: too many open files

goroutine 10 [running]:
github.com/threefoldtech/0-fs.(*G8ufs).watch(0xc00008be60)
	/gopath/src/github.com/threefoldtech/0-fs/g8ufs.go:205 +0x30d
created by github.com/threefoldtech/0-fs.Mount.func1
	/gopath/src/github.com/threefoldtech/0-fs/g8ufs.go:134 +0x6d
```
I am not sure if this is the global file number limit on all processes. or the number of files for this process. Anyway i checked the other working 0-fs processes and they have limits as follows

```
Limit                     Soft Limit           Hard Limit           Units     
Max cpu time              unlimited            unlimited            seconds   
Max file size             unlimited            unlimited            bytes     
Max data size             unlimited            unlimited            bytes     
Max stack size            8388608              unlimited            bytes     
Max core file size        0                    unlimited            bytes     
Max resident set          unlimited            unlimited            bytes     
Max processes             63325                63325                processes 
Max open files            1024                 4096                 files     
Max locked memory         65536                65536                bytes     
Max address space         unlimited            unlimited            bytes     
Max file locks            unlimited            unlimited            locks     
Max pending signals       63325                63325                signals   
Max msgqueue size         819200               819200               bytes     
Max nice priority         0                    0                    
Max realtime priority     0                    0                    
Max realtime timeout      unlimited            unlimited            us        
```

While this is still small limit for number of files, but it's probably enough.

--- a/etc/sysctl.conf
+++ b/etc/sysctl.conf
@@ -1,0 +1,2 @@
+fs.inotify.max_user_instances = 8192
+fs.inotify.max_user_watches = 524288

--- a/etc/zinit/sysctl.yaml
+++ b/etc/zinit/sysctl.yaml
@@ -1,0 +1,2 @@
+exec: sysctl -p
+oneshot: true

--- a/pkg/flist/flist.go
+++ b/pkg/flist/flist.go
@@ -176,6 +176,8 @@ func (f *flistModule) mount(name, url, storage string, opts pkg.MountOptions) (s
 		// if everything is already in place, just early return
 		log.Info().Msgf("flist is already mounted at %s, nothing more to do", mountpoint)
 		return mountpoint, nil
+	} else if err != nil {
+		return "", err
 	}
 
 	env, err := environment.Get()

--- a/qemu/overlay/etc/sysctl.conf
+++ b/qemu/overlay/etc/sysctl.conf
@@ -1,0 +1,1 @@
+../../../etc/sysctl.conf

--- a/qemu/overlay/etc/zinit/sysctl.yaml
+++ b/qemu/overlay/etc/zinit/sysctl.yaml
@@ -1,0 +1,1 @@
+../../../../etc/zinit/sysctl.yaml

